### PR TITLE
MOC-99: Simple but functional Dnd component

### PR DESCRIPTION
**Pull request by fitzk**
## Summary by CodeRabbit

- **New Features**
	- Introduced a `Draggable` component that allows users to click and drag elements around the screen.
	- Enhanced the `Popup` component with a `children` prop for flexible content placement.

- **Style**
	- Adjusted styling for the `ButtonIconSmall` component to improve layout and visual consistency.
	- Updated the `Divider` component with a class for improved styling and reordering of class names.
	- Refined the layout and structure of the `Footer` component for better responsiveness.
	- Increased the height of the `Header` component for improved visual layout.
	- Enhanced margins in the `Form` component for better vertical spacing.
	- Improved the Toast component's layout and spacing with additional utility classes.

### CODE REVIEW

File: apps/mocksi-lite/common/Popup/index.tsx [hunk 1]
```diff
@@ -1,54 +1,57 @@
+import { createSnapModifier, restrictToWindowEdges } from "@dnd-kit/modifiers";
```
Comment (elg0nz) [inline apps/mocksi-lite/common/Popup/index.tsx:1]: don't forget to explicitly add this dependency to the package.json on `mocksi-lite` and then do a `pnpm I` to update the lock file :)

**Review by elg0nz — COMMENTED** [review]
Looking good!

File: apps/mocksi-lite/common/Popup/index.tsx (hunk 1 above)
Comment (fitzk) [reply apps/mocksi-lite/common/Popup/index.tsx:1]: not using it just forgot to remove the import thanks!

File: apps/mocksi-lite/common/Popup/Draggable.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	const onMouseMove: MouseEventHandler<HTMLButtonElement | HTMLDivElement> = (
+		event,
+	) => {
+		if (pressed) {
+			setPosition({
+				x: position.x + event.movementX,
+				y: position.y + event.movementY,
+			});
+		}
+	};
```
[the remark this answers is not in the document]
Comment (fitzk) [reply apps/mocksi-lite/common/Popup/Draggable.tsx]: wrapped in useCallback to cache the fn and renamed fyi

File: apps/mocksi-lite/common/Popup/Draggable.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	const dragElRef = React.useRef<HTMLDivElement>(null);
+	const lastSavedPosition = React.useRef(position);
+
+	const persistPosition = React.useCallback(() => {
+		if (position.x !== 0 && position.y !== 0) {
+			if (
+				lastSavedPosition.current.x !== position.x ||
+				lastSavedPosition.current.y !== position.y
+			) {
+				chrome.storage.local.set({
```
Comment (elg0nz) [inline apps/mocksi-lite/common/Popup/Draggable.tsx:24]: note: I recommend adding the debounce and/or `await`ing for this to set. The reason why is that if people move their mouse too fast, there could be some race conditions where the value stored is not the last one that was set.

**Review by elg0nz — COMMENTED** [review]
just one note about debouncing

File: apps/mocksi-lite/common/Popup/Draggable.tsx (hunk 3 above)
Comment (fitzk) [reply apps/mocksi-lite/common/Popup/Draggable.tsx]: I awaited this and debounced the fn that updates the react state

**Review by elg0nz — APPROVED** [review]
Amazing!
🚢 it

**Pull request merged**